src/core/FileWatcher.cpp

```diff
@@ -43,6 +43,13 @@ FileWatcher::FileWatcher(QObject* parent)
     m_fileIgnoreDelayTimer.setSingleShot(true);
 }
 
+FileWatcher::~FileWatcher()
+{
+    // Prevent destroying the file watcher while performing async operations
+    m_mutex.lock();
+    m_mutex.unlock();
```

Comment:
Mind blown... :boom: 🤯

Reply:
Curious, do you know a better way to prevent destruction?

Reply:
Except for preventing that the object itself goes out of scope? No.

Reply:
We actively delete it in the database releaseData function!